File: Semester 4/Python Lab/EXP9/Area.py
"""Write a Python program to declare a class Calculate which calculates the Area of Circle,
Triangle and Rectangle (Use Method Overloading)."""


class Calculate:
  def area(self, **kwargs):
    if "radius" in kwargs:
      return self.area_circle(**kwargs)

    elif "length" in kwargs and "breadth" in kwargs:  
      return self.area_rectangle(**kwargs)

    elif "base" in kwargs and "height" in kwargs:
      return self.area_triangle(**kwargs)

    else:
      return "Invalid arguments"

  def area_circle(self, **kwargs):
    radius = kwargs["radius"]
    return 3.14 * radius * radius

  def area_rectangle(self, **kwargs):
    length = kwargs["length"]
    breadth = kwargs["breadth"]
    return length * breadth

  def area_triangle(self, **kwargs):
    base = kwargs["base"]
    height = kwargs["height"]
    return 0.5 * base * height

calculator = Calculate()
        
circle_area = calculator.area(radius=5)
print("\nArea of circle:", circle_area)

triangle_area = calculator.area(base=3, height=4)
print("\nArea of triangle:", triangle_area)

rectangle_area = calculator.area(length=4, breadth=6)
print("\nArea of rectangle:", rectangle_area)

File: Semester 3/Java/Q15.Write a program to calculate area of given object using method overloading/Area.java
import java.util.Scanner;

public class Area
{
    public static void main(String[] args)
    {
        System.out.println("+--------MENU-------+ ");
        System.out.println("|  1.Rectangle      | ");
        System.out.println("|  2.Circle         | ");
        System.out.println("|  3.Triange        | ");
        System.out.println("+-------------------+ ");

        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the Choice: ");
        int choice = sc.nextInt();

        switch(choice)
        {
            case 1:
            System.out.print("Enter the length of Rectangle: ");
            double length= sc.nextDouble();

            System.out.print("Enter the breadth of Rectangle: ");
            double breadth= sc.nextDouble();

            double rectangle = rectangle(length, breadth);

            System.out.println("Area of rectangle with "+length +" and "+ breadth + " is " + rectangle);

            break;

            case 2:
            System.out.print("Enter the radius of circle: ");
            double radius= sc.nextDouble();

            double circle = circle(radius);

            System.out.println("Area od Circle with "+ radius + " is "+ circle);

            break;

            case 3:
            System.out.print("Enter the base of Triangle: ");
            double base= sc.nextDouble();

            System.out.print("Enter the height of Triangle: ");
            double height= sc.nextDouble();

            double triangle = triangle(base, height);
            System.out.println("Area of Triangle with "+ base + " and " + height + " is "+ triangle);
            break;

            default:
             System.out.print("Enter Valid Choice!! ");
            break;


            


        }

            sc.close();

        


    }

    public static double rectangle(double length,double breadth)
    {
        return length * breadth;
    }

    public static double circle(double radius)
    {
        return Math.PI * radius *radius;
    }

    public static double triangle(double base,double height)
    {
        return 0.5*base*height;
    }
}